tests/Shared/TemplatesTesting/BuildEnvironment.cs

```diff
@@ -21,11 +21,14 @@ public class BuildEnvironment
     public DirectoryInfo?                   RepoRoot                      { get; init; }
     public TemplatesCustomHive?             TemplatesCustomHive           { get; init; }
 
-    public static readonly string TempDir = IsRunningOnCI
-        ? Path.GetTempPath()
-        : Environment.GetEnvironmentVariable("DEV_TEMP") is { } devTemp && Path.Exists(devTemp)
-            ? devTemp
-            : Path.GetTempPath();
+    public static readonly string TempDir =
```

Comment:
Could you please add a comment explaining why we're doing it and all the scenarios it covers.